guacamole/src/main/java/org/glyptodon/guacamole/net/basic/rest/auth/TokenRESTService.java

```diff
@@ -151,8 +161,13 @@ private String getLoggableAddress(HttpServletRequest request) {
     public APIAuthToken createToken(@FormParam("username") String username,
             @FormParam("password") String password,
             @FormParam("token") String token,
-            @Context HttpServletRequest request) throws GuacamoleException {
+            @Context HttpServletRequest consumedRequest,
+            MultivaluedMap<String, String> parameters)
+            throws GuacamoleException {
 
+        // Reconstitute the HTTP request with the map of parameters
+        HttpServletRequest request = new APIRequest(consumedRequest, parameters);
```

Comment:
The same way that we are able to obtain other parameters through annotations. The request has been consumed, yes, but by JAX-RS, and it's JAX-RS that provides the ```MultivaluedMap```.